python/syntax.lark

```diff
@@ -141,7 +148,7 @@ func_type: generic_declaration? (func_inner_type "->")+ not_func_type
 
 ?number_expression: sum_expression
 ?sum_expression: [sum_expression (ADD | SUBTRACT)] product_expression
-?product_expression: [product_expression (MULTIPLY | DIVIDE | MODULO | SHIFTL | SHIFTR)] exponent_expression
+?product_expression: [product_expression (MULTIPLY | DIVIDE | MODULO | SHIFTL | IN | SHIFTR)] exponent_expression
```

Comment:
Oh, very interesting, you made it at the same precedence level as multiplication? So `a * b in c / 3` would be the same as `((a * b) in c) / 3`, and `"wow" + "ok" in "lolwowokwee"` would be a type error

It seems in JavaScript and Python, `in` is at the same precedence level as the comparison operators `>` `<=` etc. This makes sense because `in` is a boolean operator. We could consider allowing chaining expressions such as `"a" in "apple" in "banana and apple cider"`. We don't have to do that in this PR, but if we want to add something like this in the future, we'll have to make sure we can do so without breaking compatibility

Reply:
OH wait oops